### PR TITLE
Status effects are actually deleted/handled when the mob is deleted

### PR DESCRIPTION
**Pull request by ChangelingRain**
a thing @Robustin needs, actually,

### CODE REVIEW

**Comment by Robustin** [comment]
Now I just need to figure out exactly how status effects work so that I can have a custom one for the cult master.

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -29,6 +29,13 @@
 	med_hud_set_status()
 
 /mob/living/Destroy()
+	if(LAZYLEN(status_effects))
+		for(var/s in status_effects)
+			var/datum/status_effect/S = s
+			if(S.on_remove_on_mob_delete) //the status effect calls on_remove when its mob is deleted
+				qdel(S)
+			else
+				S.be_replaced()
```
Comment (AnturK) [inline code/modules/mob/living/living.dm]: Why ?

Comment (ChangelingRain) [inline code/modules/mob/living/living.dm]: It handles the stuff we need minus it actually calling on_remove()

Comment (AnturK) [inline code/modules/mob/living/living.dm]: It's just i don't feel the name of the function represents the use well if it's now the default removal proc.

**Pull request merged**